server/nodeset/upload-deposit-data.go

```diff
@@ -68,79 +69,109 @@ func (c *nodesetUploadDepositDataContext) PrepareData(data *swapi.NodesetUploadD
 		return types.ResponseStatus_WalletNotReady, err
 	}
 
-	// Get the list of registered validators
-	registeredPubkeyMap := map[beacon.ValidatorPubkey]bool{}
-	pubkeyStatusResponse, err := nc.GetRegisteredValidators(ctx)
+	// Fetch private keys and derive public keys
+	privateKeys, err := w.GetAllPrivateKeys()
```

Comment:
This is kind of an expensive process so you want to defer it as long as possible; that's why the nodeset-client query was done first previously.

Reply:
resolved